src/utils/EnumerableAddressSetMap.sol

```diff
@@ -1,186 +1,186 @@
+                        sstore(keccak256(0x0c, 0x34), valueIndex) 
+                    } 
+                } 
+                // Delete the slot where the moved value was stored 
+                currValues.pop(); 
+ 
+                /// @solidity memory-safe-assembly 
+                assembly { 
+                    // Delete the index for the deleted slot 
+                    // Equivalent to: 
+                    // `_indexes[key][value] = 0`. 
+                    sstore(valueSlot, 0) 
+                } 
+            } 
+        } 
+    } 
+ 
+    // ============================================================= 
+    //                        VIEW FUNCTIONS 
+    // ============================================================= 
+ 
+    /** 
+     * @dev Returns whether `value` is in the enumerable address set 
+     *      at `key` in `map`. 
+     * @param map   The mapping of enumerable address sets. 
+     * @param key   The key into the mapping. 
+     * @param value The value to check. 
+     * @return result The result. 
+     */ 
+    function contains(Map storage map, address key, address value) 
+        internal 
+        view 
+        returns (bool result) 
+    { 
+        /// @solidity memory-safe-assembly 
+        assembly { 
+            // The value is stored at length-1, but we add 1 to all indexes 
+            // and use 0 as a sentinel value. 
+            // Equivalent to: 
+            // `result = _indexes[key][lastValue] != 0`. 
+            mstore(0x20, value) 
+            mstore(0x0c, map.slot) 
+            mstore(returndatasize(), key) 
+            result := iszero(iszero(sload(keccak256(0x0c, 0x34)))) 
+        } 
+    } 
+ 
+    /** 
+     * @dev Returns the length of the enumerable address set 
+     *      at `key` in `map`. 
+     * @param map The mapping of enumerable address sets. 
+     * @param key The key into the mapping. 
+     * @return The length. 
+     */ 
+    function length(Map storage map, address key) internal view returns (uint256) { 
+        return map._values[key].length; 
+    } 
+ 
+    /** 
+     * @dev Returns the value at `index` of the enumerable address set 
+     *      at `key` in `map`. 
+     * @param map   The mapping of enumerable address sets. 
+     * @param key   The key into the mapping. 
+     * @param index The index of the enumerable address set. 
+     * @return The value. 
+     */ 
+    function at(Map storage map, address key, uint256 index) internal view returns (address) { 
+        return map._values[key][index]; 
+    } 
+ 
+    /** 
+     * @dev Returns all the values of the enumerable address set 
+     *      at `key` in `map`. 
+     * @param map The mapping of enumerable address sets. 
+     * @param key The key into the mapping. 
+     * @return The values. 
+     */ 
+    function values(Map storage map, address key) internal view returns (address[] memory) { 
+        return map._values[key]; 
+    } 
+} 
```

Comment:
I feel like it would be safer to make this lib a separate PR with a diff against the OZ contract for easier reviewing.

However, if everyone else is comfortable with including here, ignore my comment.

Reply:
This lib is specifically only for this registry, as it abuses `returndatasize()` instead of `0`.

This trick can only be used if the entire contract does not make any external calls or staticcalls.